example-ideal.txt

```diff
@@ -0,0 +1,22 @@
+# A Picobot program that can fill an empty room
+
+# move to the top left
+
+moveLeft:
+	If can move left, move left   # go all the way to the left
+	If cannot move left, moveUp   # can't go left anymore, so try to go up
+
+moveUp:
+	If can move up, move up                                             # go all the way to the top
+	If cannot move up and can move down, move down and fillColumnDown   # can't go up any more, so try to go down
+
+
+# fill from top to bottom, left to right
+
+fillColumnDown:   # fill this column to the bottom
+	If can move down, move down                                           # go all the way to the bottom
+	If cannot move down but can move right, move right and fillColumnUp   # can't go down anymore, so try to go right
+
+fillColumnUp:     # fill this column to the top
+	If can move up, move up                                               # go all the way to the top
```

Comment:
I love the clarity of the state names. They are so much more readable than the numbers states.